api/dao/base.py

```diff
@@ -80,12 +80,17 @@ def get_container(self, _id, projection=None, get_children=False):
             cont[CHILD_MAP[self.cont_name]] = children
         return cont
 
-    def get_children(self, _id, projection=None):
+    def get_children(self, _id, projection=None, uid=None):
         try:
             child_name = CHILD_MAP[self.cont_name]
         except KeyError:
             raise APINotFoundException('Children cannot be listed from the {0} level'.format(self.cont_name))
-        query = {self.cont_name[:-1]: bson.ObjectId(_id)}
+        if self.cont_name == 'groups':
+            query = {self.cont_name[:-1]: _id}
```

Comment:
We have a utility method in `contianerutil.py` that singularizes container names, mind changing this existing code to use that instead? Analysis -> Analyses ruined the "remove the s" hack (also wouldn't affect us here since analyses don't have children but good to clean up that old hack when we can).